packages/console/src/components/ApplicationName/index.tsx

```diff
@@ -16,13 +19,38 @@ type Props = {
 function ApplicationName({ applicationId, isLink = false }: Props) {
   const isAdminConsole = applicationId === adminConsoleApplicationId;
 
-  const { data } = useSWR<Application>(!isAdminConsole && `api/applications/${applicationId}`);
+  const fetchApi = useApi({ hideErrorToast: true });
+  const fetcher = useSwrFetcher<Application>(fetchApi);
+  const { data, error } = useSWR<Application, RequestError>(
+    !isAdminConsole && `api/applications/${applicationId}`,
+    {
+      fetcher,
+      shouldRetryOnError: (error: unknown) => {
+        if (error instanceof RequestError) {
+          return error.status !== 404;
+        }
+
+        return true;
```

Comment:
i found similar code:

<img width="353" alt="image" src="https://github.com/logto-io/logto/assets/14722250/62045697-89fb-4727-be41-8481296c119f">

can we make a util for ignore certain status codes?

Reply:
ok

Reply:
Updated